base/random.jl

```diff
@@ -1166,6 +1166,29 @@ immutable UUID
 end
 UUID(u::AbstractString) = convert(UUID, u)
```

Comment:
It's just that the default constructor (which takes one arg and tries to convert it to a `UInt128`) is taking precedence, so this method is indeed needed.